src/psyclone/gocean1p0.py

```diff
@@ -700,8 +678,10 @@ def _upper_bound(self):
                                    self)
             # TODO 363 - needs to be updated once the PSyIR has support for
             # Fortran derived types.
-            stop.addchild(Literal(self.field_name+"%data", DataType.INTEGER,
-                                  parent=stop))
+            api_config = Config.get().api_conf("gocean1.0")
+            data = api_config.field_properties["go_grid_data"][0] \
```

Comment:
Please could you add a comment here (something along the lines of "Look-up the name of the array containing the grid points").

Reply:
I've used somewhat different wording, hope that's ok.